distributed/scheduler.py

```diff
@@ -4760,12 +4761,8 @@ async def handle_worker_status_change(
                 worker_msgs: dict = {}
                 self._transitions(recs, client_msgs, worker_msgs, stimulus_id)
                 self.send_all(client_msgs, worker_msgs)
-        elif ws.status == Status.paused:
-            self.running.remove(ws)
-        elif ws.status == Status.closing:
-            await self.remove_worker(
-                address=ws.address, stimulus_id=stimulus_id, close=False
-            )
+        else:
+            self.running.discard(ws)
```

Comment:
Resolves https://github.com/dask/distributed/pull/6363/files#r878475185

This is equivalent to main before the PR: https://github.com/dask/distributed/blob/41a54ee09861b9d531501c7518021fc6213b263c/distributed/scheduler.py#L4747-L4777